moped-editor/src/views/staff/StaffListView.js

```diff
@@ -1,73 +1,112 @@
 
 const StaffListView = () => {
   const classes = useStyles();
 
-  const [showInactive, setShowInactive] = useState(false);
+  const { data, loading, error } = useQuery(GET_ALL_USERS);
+
+  const staffColumns = [
+    {
+      headerName: "",
+      field: "user_id",
+      renderCell: (props) => <EditUserButton id={props.value} />,
+      width: 50,
+      sortable: false,
+      filterable: false,
+    },
+    {
+      headerName: "First name",
+      field: "first_name",
+      width: 125,
+    },
+    {
+      headerName: "Last name",
+      field: "last_name",
+      width: 125,
+    },
+    {
+      headerName: "E-mail",
+      field: "email",
+      width: 325,
+    },
+    {
+      headerName: "Title",
+      field: "title",
+      width: 250,
+    },
+    {
+      headerName: "Workgroup",
+      field: "moped_workgroup",
+      valueGetter: (props) => props.row.moped_workgroup.workgroup_name,
+      width: 300,
+    },
+    {
+      headerName: "Role",
+      field: "roles",
+      renderCell: (props) => {
+        if (!props.value || !props.value[0]) {
+          return "N/A";
+        }
 
-  /**
-   * Override 'where' key based on state of 'showInactive'
-   * @type {GQLAbstract}
-   */
-  const staffQuery = new GQLAbstract({
-    ...StaffListViewQueryConf,
-    ...(showInactive ? { where: {} } : {}),
-  });
+        const role = props.value[0].replace("moped-", "");
+        return role.charAt(0).toUpperCase() + role.slice(1);
+      },
+    },
+    {
+      headerName: "Is deleted",
+      field: "is_deleted",
+      valueGetter: (props) => (props.value ? "True" : "False"),
+    },
+  ];
 
-  /**
-   * Toggles list of inactive users
-   */
-  const toggleShowInactive = () => {
-    setShowInactive(!showInactive);
-  };
+  if (error) {
+    console.error(error);
+  }
 
   return (
     <Page className={classes.root} title="Staff">
-      <GridTable
-        title={"Staff"}
-        query={staffQuery}
-        toolbar={null}
-        customComponents={{
-          table: {
-            before: (
-              <FormGroup className={classes.switch}>
-                <FormControlLabel
-                  control={
-                    <Switch
-                      checked={showInactive}
-                      onChange={toggleShowInactive}
-                      color={"primary"}
-                    />
-                  }
-                  labelPlacement="start"
-                  label="Show Inactive Accounts"
-                />
-              </FormGroup>
-            ),
-          },
-        }}
-      />
+      {loading ? (
+        <CircularProgress />
+      ) : (
+        <Container className={classes.container}>
+          <Box className={classes.addStaffButton}>
+            <AddUserButton />
+          </Box>
+          <Card>
+            <DataGrid
+              rows={data["moped_users"]}
+              columns={staffColumns}
+              getRowId={(row) => row.user_id}
+              slots={{ toolbar: GridToolbar }}
```

Comment:
when i tested out the CSV download it only returned one row, then i realized i had accidentally selected a row by clicking on it, and the csv only returned that selected row. is there a prop to disable the row selecting altogether?

Reply:
yayyyy 🙌